qiime2/sdk/result.py

```diff
@@ -143,9 +143,40 @@ def export_data(self, output_dir):
     def _destructor(self):
         return self._archiver._destructor
 
-    def save(self, filepath):
-        if not filepath.endswith(self.extension):
-            filepath += self.extension
+    def save(self, filepath, ext=None):
+        """Save to a file.
+
+        Parameters
+        ----------
+        filepath : str
+            Path to save file at.
+
+        extension : str
+            Preferred file extension (.qza, .qzv, .txt, etc).
+            If no preferred extension input is included,
+            Artifact extension will default to .qza and
+            Visualization extension will default to .qzv.
+            Including a period in the extension is
+            optional, and any additional periods delimiting
+            the filepath and the extension will be reduced
+            to a single period.
+
```

Comment:
This is missing the 'Returns' section:

```suggestion

        Returns
        -------
        str
            Filepath and extension (if provided) that the
            file was saved to.
```